apps/job-service/src/jobs/delayed-jobs.controller.ts

```diff
@@ -11,28 +12,78 @@ import { DelayedJob } from '@terramatch-microservices/database/entities';
 
 @Controller('jobs/v3/delayedJobs')
 export class DelayedJobsController {
+  @Get()
+  @ApiOperation({
+    operationId: 'listDelayedJobs',
+    description: 'Retrieve a list of all delayed jobs.',
+  })
+  @JsonApiResponse({ data: { type: DelayedJobDto } })
+  @ApiException(() => UnauthorizedException, {
+    description: 'Authentication failed.',
+  })
+  async getRunningJobs(
+    @Request() { authenticatedUserId }
+  ): Promise<JsonApiDocument> {
+    const runningJobs = await DelayedJob.findAll({
+      where: {
+        isCleared: false,
+        createdBy: authenticatedUserId
+      },
+      order: [['createdAt', 'DESC']],
+    });
+
+    const document = buildJsonApi();
+    runningJobs.forEach((job) => {
+      document.addData(job.uuid, new DelayedJobDto(job));
+    });
+    return document.serialize();
+  }
+
   @Get(':uuid')
   @ApiOperation({
     operationId: 'delayedJobsFind',
-    description: 'Get the current status and potentially payload or error from a delayed job.',
+    description:
+      'Get the current status and potentially payload or error from a delayed job.',
   })
   @JsonApiResponse({ data: { type: DelayedJobDto } })
   @ApiException(() => UnauthorizedException, {
     description: 'Authentication failed.',
   })
   @ApiException(() => NotFoundException, {
-    description: 'Job with that UUID not found.'
+    description: 'Job with that UUID not found.',
   })
+  // Note: Since jobs are very generic and we don't track which resources are related to a given
+  // job, there is no effective way to make a policy for jobs until we expand the service to
+  // include an owner ID on the job table.
   async findOne(@Param('uuid') pathUUID: string): Promise<JsonApiDocument> {
-    const job = await DelayedJob.findOne({ where: { uuid: pathUUID }});
+    const job = await DelayedJob.findOne({ where: { uuid: pathUUID } });
     if (job == null) throw new NotFoundException();
 
-    // Note: Since jobs are very generic and we don't track which resources are related to a given
-    // job, there is no effective way to make a policy for jobs until we expand the service to
-    // include an owner ID on the job table.
-
     return buildJsonApi()
       .addData(pathUUID, new DelayedJobDto(job))
       .document.serialize();
   }
-}
+
+  @Patch('clear')
+  @ApiOperation({
+    operationId: 'clearNonPendingJobs',
+    description: 'Set isCleared to true for all jobs where status is not pending.',
+  })
+  @ApiException(() => UnauthorizedException, {
+    description: 'Authentication failed.',
+  })
+  async clearNonPendingJobs(@Request() { authenticatedUserId }): Promise<{ message: string }> {
+    const updatedCount = await DelayedJob.update(
+      { isCleared: true },
+      {
+        where: {
+          isCleared: false,
+          status: { [Op.ne]: 'pending' },
+          createdBy: authenticatedUserId,
+        },
+      }
+    );
+
+    return { message: `${updatedCount[0]} jobs have been cleared.` };
```

Comment:
No endpoint in v3 should be sending a plain JSON response like this. If a response is required, it must be in JsonAPI format. In the case of a Patch operation, the expected return value would be the modified resource(s), which the FE Connection system will know how to fold into its cache of the server data so that those resources will be updated client side and any component that relies on them will get automatically updated with the refreshed data.

Note that in the case of the research service, I left off the response value. I possibly shouldn't have done that, but that API is different, not being consumed by our FE API layer.

Reply:
the response is now the modified resources